tests/test_realvect.py

```diff
@@ -0,0 +1,292 @@
+def test_3d_max():
+    # max
+    v1 = amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)])
+    v2 = amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)])
+    ref_list = [6,3,6]
+    compare_vect = amrex.RealVect([ref_list[ii] for ii in range(amrex.Config.spacedim)])    
+    v3 = amrex.max(v1,v2)
+    assert(v3 == compare_vect)
+    assert(v1 == amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)]))
+    assert(v2 == amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)]))
+
+    v1.max(v2)
+    assert(v1 == compare_vect)
+    assert(v2 == amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)]))
+
+    v1 = amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)])
+    v2 = amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)])
+
+    amrex.RealVect.max(v2,v1)
+    assert(v2 == compare_vect)
+    assert(v1 == amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)]))
+
+@pytest.mark.skipif(amrex.Config.spacedim != 3,
+                    reason="Requires AMREX_SPACEDIM = 3")
+def test_3d_min():
+    # min
+    v1 = amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)])
+    v2 = amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)])
+    ref_list = [0,3,0]
+    compare_vect = amrex.RealVect([ref_list[ii] for ii in range(amrex.Config.spacedim)])
+    v3 = amrex.min(v1,v2)
+    assert(v3 == compare_vect)
+    assert(v1 == amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)]))
+    assert(v2 == amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)]))
+
+    v1.min(v2)
+    assert(v1 == compare_vect)
+    assert(v2 == amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)]))
+
+    v1 = amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)])
+    v2 = amrex.RealVect([3*(2-ii) for ii in range(amrex.Config.spacedim)])
+
+    amrex.RealVect.min(v2,v1)
+    assert(v2 == compare_vect)
+    assert(v1 == amrex.RealVect([3*ii for ii in range(amrex.Config.spacedim)]))
+
+
+@pytest.mark.skipif(amrex.Config.spacedim != 3,
+                    reason="Requires AMREX_SPACEDIM = 3")
+def test_properties():
+    tl = [1,-5,4.2]
+    v1 = amrex.RealVect(tl)
+    
+    # sum
+    assert(np.isclose(v1.sum, sum(tl) ))
+    # vectorLength
+    assert(np.isclose(v1.vectorLength, np.linalg.norm(tl)))
+    # // radSquared
+    assert(np.isclose(v1.radSquared, np.linalg.norm(tl)**2))
+    # // product
+    assert(np.isclose(v1.product, np.prod(tl)))
+    # // minDir
+    assert(v1.minDir(False) == 1)
+    assert(v1.minDir(True) == 0)
+    # // maxDir
+    assert(v1.maxDir(False) == 2)
+    assert(v1.maxDir(True) == 1)
+
+def test_basis_vector():
+    for ii in range(amrex.Config.spacedim):
+        ei = amrex.RealVect.BASISREALV(ii)
+        for jj in range(amrex.Config.spacedim):
+            if ii == jj:
+                assert(ei[jj] == 1)
+            else:
+                assert(ei[jj] == 0)
+
+
+
+
+
+
+
+
+
```

Comment:
```suggestion
```

Reply:
That was a lot of white space to eliminate!